### PR TITLE
Muzammil 233 project sub teams

### PR DESCRIPTION
**Pull request by MuzammilC**
This ticket entailed the addition of nested sub-team pages to a project.

## Updates

[Please enter a brief summary of updates to this PR during code review - if any]

## Checklist

- [x] Ran `npm run lint:fix` to format code
- [x] Requested at least one reviewer
- [x] Connected PR to Trello ticket (steps below)
- [ ] Addressed code review comments
- [ ] Gained at least one approval for your PR

## Connecting your PR to the corresponding ticket:
- Click on the "GitHub" button found on the right of your Trello ticket
- Choose "Attach PR" from the dropdown list
- Link your GitHub account (if you haven't done so already)
- Navigate to`VCL-content-platform` and choose your PR from the dropdown.

### CODE REVIEW

File: src/pages/ProjectV2/Wrapper/ProjectV2Wrapper.tsx [hunk 1]
```diff
@@ -19,6 +22,24 @@ const ProjectV2Wrapper: React.FC<ProjectProps> = ({ match }) => {
     const [openMobileDropdown, setOpenMobileDropdown] = useState<boolean>(false);
     const dispatch = useAppDispatch();
 
+
+    const currentProject = PROJECT_TEXT.find((project) => project.project_title === match.params.project_id);
+    const subPages : Array<string> = [];
+    const [isSubpage,setIsSubpage] = useState<boolean>(false);
+
+    
+    if (currentProject && currentProject.subpages) {
```
Comment (ryanyae) [inline src/pages/ProjectV2/Wrapper/ProjectV2Wrapper.tsx:31]: If you really want to clean this up you can just check if a subpages property exists inside the ProjectRouter and conditionally render the subpages router

File: src/pages/ProjectV2/Wrapper/ProjectV2Wrapper.tsx [hunk 2]
```diff
@@ -72,6 +97,16 @@ const ProjectV2Wrapper: React.FC<ProjectProps> = ({ match }) => {
                                 <div className="single-table-content">
                                     <a className="project-content-container-href" href={`/projectsV2/${match.params.project_id}/`}>Home</a>
                                 </div>
+                                {subPages &&
```
Comment (ryanyae) [inline src/pages/ProjectV2/Wrapper/ProjectV2Wrapper.tsx:100]: Could also do this here, but the suggestion I gave isnt completely necessary just maybe a little more readable. But my approach will take away the universal aspect of the subpages but idk if that really matters here.

File: src/pages/ProjectV2/Wrapper/ProjectV2Wrapper.tsx [hunk 3]
```diff
@@ -72,6 +97,16 @@ const ProjectV2Wrapper: React.FC<ProjectProps> = ({ match }) => {
...
+                                    subPages.map((subpageTitle: string) => {
+                                        return(
+                                            <div className="single-table-content" key={subpageTitle}>
```
Comment (ryanyae) [inline src/pages/ProjectV2/Wrapper/ProjectV2Wrapper.tsx:103]: You should add indents to these subpage redirections on the sidebar to indicate that there are 'subpages'

**Review by ryanyae — APPROVED** [review]
Just consider my recommendations and give me a reply if you have made a decision. I think this implementation is good already and doesnt really need fixing.

**Comment by MuzammilC** [comment]
[projectV2 Adding sub-teams to project sidebar](https://trello.com/c/VcuJSSj3/233-projectv2-adding-sub-teams-to-project-sidebar)

File: src/pages/ProjectV2/Wrapper/ProjectV2Wrapper.tsx (hunk 1 above)
Comment (MuzammilC) [reply src/pages/ProjectV2/Wrapper/ProjectV2Wrapper.tsx:31]: But isn't the ProjectRouter already rendering the subpages router conditionally, based on the isPages prop?

File: src/pages/ProjectV2/Wrapper/ProjectV2Wrapper.tsx (hunk 3 above)
Comment (MuzammilC) [reply src/pages/ProjectV2/Wrapper/ProjectV2Wrapper.tsx:103]: Will do that

File: src/pages/ProjectV2/Wrapper/ProjectV2Wrapper.tsx (hunk 2 above)
Comment (MuzammilC) [reply src/pages/ProjectV2/Wrapper/ProjectV2Wrapper.tsx:100]: I was thinking of cleaning up the code, but the way the static file is set, a cleaner version would be very similar to this, but I'm happy to get the cleaner version done if you think It'll help drastically.

**Comment by ElisaHu1** [comment]
Failed to compile.

/Users/elisahu/VCL/VCL-content-platform/src/pages/ProjectV2/Pages/Join/ProjectV2Join.tsx
TypeScript error in /Users/elisahu/VCL/VCL-content-platform/src/pages/ProjectV2/Pages/Join/ProjectV2Join.tsx(37,21):
'currentProject.joinTeam' is possibly 'undefined'.  TS18048

    35 |                 <h1>What We Do</h1>
    36 |                 {
  > 37 |                     currentProject?.joinTeam.whatWeDo.map((item, index) => {
       |                     ^
    38 | 
    39 |                         return (<p key={index}>{item}</p>)
    40 |                     })

**Review by ElisaHu1 — APPROVED** [review]

**Pull request merged**